### ISSUE TITLE
Some unit test cases doesn't work on Windows

### ISSUE THREAD

**Issue by JeyJeyGao** [notaryproject/notation#648]
### What is the areas you experience the issue in?

Notation CLI

### What is not working as expected?

Some unit test cases doesn't work on Windows

```
go: downloading github.com/notaryproject/notation-core-go v1.0.0-rc.3
go: downloading github.com/notaryproject/notation-go v1.0.0-rc.4
go: downloading github.com/spf13/cobra v1.7.0
go: downloading golang.org/x/term v0.7.0
go: downloading oras.land/oras-go/v2 v2.0.2
go: downloading golang.org/x/crypto v0.7.0
go: downloading golang.org/x/mod v0.10.0
go: downloading golang.org/x/sys v0.7.0
go: downloading github.com/golang-jwt/jwt/v4 v4.5.0
?       github.com/notaryproject/notation/cmd/notation/internal/cmdutil [no test files]
?       github.com/notaryproject/notation/internal/cmd  [no test files]
?       github.com/notaryproject/notation/internal/ioutil       [no test files]
?       github.com/notaryproject/notation/cmd/notation/internal/errors  [no test files]
?       github.com/notaryproject/notation/internal/experimental [no test files]
?       github.com/notaryproject/notation/cmd/notation/policy   [no test files]
ok      github.com/notaryproject/notation/cmd/notation  3.365s
ok      github.com/notaryproject/notation/cmd/notation/cert     4.430s
ok      github.com/notaryproject/notation/cmd/notation/internal/truststore      3.627s
ok      github.com/notaryproject/notation/internal/envelope     5.218s
--- FAIL: TestWriteFile (0.04s)
    --- FAIL: TestWriteFile/write_file_with_directory_permission_error (0.01s)
        file_test.go:40: should write failed
--- FAIL: TestWriteFileWithPermission (0.02s)
    --- FAIL: TestWriteFileWithPermission/write_with_directory_permission_error (0.01s)
        file_test.go:91: should have an error
--- FAIL: TestCopyToDir (0.07s)
    --- FAIL: TestCopyToDir/source_directory_permission_error (0.01s)
        file_test.go:141: should have error
    --- FAIL: TestCopyToDir/source_file_permission_error (0.01s)
        file_test.go:168: should have error
    --- FAIL: TestCopyToDir/dest_directory_permission_error (0.01s)
        file_test.go:187: should have error
    --- FAIL: TestCopyToDir/dest_directory_permission_error_2 (0.01s)
        file_test.go:206: should have error
FAIL
FAIL    github.com/notaryproject/notation/internal/osutil       3.466s
ok      github.com/notaryproject/notation/internal/slices       3.136s
ok      github.com/notaryproject/notation/internal/trace        3.055s
ok      github.com/notaryproject/notation/internal/tree 2.805s
ok      github.com/notaryproject/notation/internal/version      1.854s
ok      github.com/notaryproject/notation/pkg/auth      2.253s
--- FAIL: TestIsRegistryInsecureConfigPermissionError (0.00s)
    util_test.go:92: should false because of missing config.json read permission.
--- FAIL: TestResolveKey (0.00s)
    --- FAIL: TestResolveKey/signingkeys.json_without_read_permission (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x18 pc=0x1f59ae]

goroutine 20 [running]:
testing.tRunner.func1.2({0x2149c0, 0x3995a0})
        C:/Program Files/Go/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
        C:/Program Files/Go/src/testing/testing.go:1529 +0x39f
panic({0x2149c0, 0x3995a0})
        C:/Program Files/Go/src/runtime/panic.go:884 +0x213
github.com/notaryproject/notation/pkg/configutil.TestResolveKey.func4(0xc00016d1e0)
        C:/Users/junjiegao/Downloads/projects/notation/pkg/configutil/util_test.go:135 +0x12e
testing.tRunner(0xc00016d1e0, 0x266ac8)
        C:/Program Files/Go/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
        C:/Program Files/Go/src/testing/testing.go:1629 +0x3ea
FAIL    github.com/notaryproject/notation/pkg/configutil        1.487s
FAIL
```

### What did you expect to happen?

The unit test should work on both Windows and Linux.

### How can we reproduce it?

Run `go test ./...` on Windows.

### Describe your environment

Windows 11 x64

### What is the version of your Notation CLI or Notation Library?

Source code

**Comment by JeyJeyGao** [comment]
Reported by @Wwwsylvia

**Comment by JeyJeyGao** [comment]
Fixed by #651
